src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/wrapper/ErrorHandlingWrapper.java

```diff
@@ -40,15 +41,23 @@ public ErrorHandlingWrapper(ErrorHandler errorHandler) {
         }
     }
 
-    public void wrapDocUploadFailure(Envelope envelope, Supplier<Void> supplier) {
+    /**
+     * Wraps docs upload to handles related errors.
+     *
+     * @param envelope for the documents to upload
+     * @param supplier upload function
+     * @return true if upload successful
```

Comment:
atm (sadly) it return whatever the passed supplier returns.
But it always returns `false` if running the `supplier` threw an exception.

Reply:
Yes, if the upload failed should definitely return false. Otherwise the supplier should know what to return